crates/key-server/src/tests/server.rs

```diff
@@ -8,6 +8,81 @@ use tracing_test::traced_test;
 use crate::externals::get_latest_checkpoint_timestamp;
 use crate::tests::SealTestCluster;
 
+use crate::key_server;
+use crate::Network;
+use crate::PACKAGE_VERSION;
+use axum::body::Body;
+use axum::extract::Request;
+use axum::http::StatusCode;
+use crypto::ibe::generate_key_pair;
+use rand::thread_rng;
+use sui_types::base_types::ObjectID;
+use tower::{Service, ServiceExt};
+
+#[tokio::test]
+async fn test_http_response() {
+    let master_key = generate_key_pair(&mut thread_rng()).0;
+    let network = Network::Devnet;
+    let object_id = ObjectID::from_single_byte(0); // Dummy value
+
+    let mut app = key_server(master_key, network, object_id)
+        .await
+        .into_service();
+
+    // No SDK version
+    let request = Request::builder()
+        .uri("/v1/service")
+        .body(Body::empty())
+        .unwrap();
+    let response = ServiceExt::<Request<Body>>::ready(&mut app)
+        .await
+        .unwrap()
+        .call(request)
+        .await
+        .unwrap();
+    assert_eq!(
+        PACKAGE_VERSION,
+        response.headers().get("X-KeyServer-Version").unwrap()
+    );
+    assert_eq!(response.status(), StatusCode::FORBIDDEN);
```

Comment:
maybe this should be BAD_REQUEST?

Reply:
Fixed in latest commit. Unit tests are a bit more complicated due to the metrics server so they are removed for now.